### PR TITLE
Product page

### PR DESCRIPTION
**Pull request by JuvitaSaini09**
Added Product page

### CODE REVIEW

File: src/components/navbar/navbar.js [hunk 1]
```diff
@@ -1,33 +1,33 @@
...
+      <header className="header-nav">
+        <div className="navbar">
+          <div className="logo">
+            <h1><i className="fab fa-artstation" />Quartz</h1>
+          </div>
+          <nav className="nav">
+            <ul className="nav-links">
 
-      <Link to="/"><li><a href="/">HOME</a></li></Link>
+              <Link to="/"><li><a href="/">HOME</a></li></Link>
```
Comment (Muzammil-khan710) [inline src/components/navbar/navbar.js:14]: Remove a tag as you are already using Link tag

File: src/components/navbar/navbarLogout.js [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
+            <h1><i className="fab fa-artstation" />Quartz</h1>
+          </div>
+          <nav className="nav">
+            <ul className="nav-links">
 
-      <Link to="/"><li><a href="/index.html">HOME</a></li></Link>
+              <Link to="/"><li><a href="/index.html">HOME</a></li></Link>
 
-    <Link to="/ProductPage"><li><a href="/components/product-page/product-page.html">PRODUCTS</a></li></Link>   
+              <Link to="/ProductPage"><li><a href="/components/product-page/product-page.html">PRODUCTS</a></li></Link>
```
Comment (Muzammil-khan710) [inline src/components/navbar/navbarLogout.js:16]: Same here

**Review by Muzammil-khan710 — APPROVED** [review]
Code is readable and well structured good to merge !

File: src/App.js [hunk 3]
```diff
@@ -2,13 +2,15 @@ import { Route,Routes } from "react-router-dom";
...
 import { Home } from "./pages/home/home";
+import { ProductPage } from "./pages/productPage/productPage";
 
 function App() {
   return (
     <div className="App">
        
         <Routes>
             <Route path="/"  element={<Home />} />
+            <Route path="/ProductPage"  element={<ProductPage />} />
```
Comment (kgajriaAgarwal) [inline src/App.js:13]: inspite of "/ProductPage" you can use "/products"

File: src/components/navbar/navbar.js [hunk 4]
```diff
@@ -1,33 +1,33 @@
...
 
-    <Link to="/ProductPage"><li><a href="/" >PRODUCTS</a></li></Link>   
+              <Link to="/ProductPage"><li><a href="/" >PRODUCTS</a></li></Link>
```
Comment (kgajriaAgarwal) [inline src/components/navbar/navbar.js:16]: inspite of a tag use Link or NavLink

File: src/components/navbar/navbarLogout.js [hunk 5]
```diff
@@ -1,33 +1,33 @@
...
+      <header className="header-nav">
+        <div className="navbar">
+          <div className="logo">
+            <h1><i className="fab fa-artstation" />Quartz</h1>
+          </div>
+          <nav className="nav">
+            <ul className="nav-links">
 
-      <Link to="/"><li><a href="/index.html">HOME</a></li></Link>
+              <Link to="/"><li><a href="/index.html">HOME</a></li></Link>
```
Comment (kgajriaAgarwal) [inline src/components/navbar/navbarLogout.js:14]: inspite of a tag use Link or NavLink

File: src/components/productListing/productListing.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
+function ProductListing() {
+  const { state } = useFilter();
+
+  return (
+    <section className="product-page" style={{ marginTop: 3 + "rem" }}>
+
+      {!state.length ? <h2 style={{ marginTop: 10 + "rem", marginLeft: 10 + "rem" }}>No Match Found !!</h2> : ""}
```
Comment (kgajriaAgarwal) [inline src/components/productListing/productListing.js:12]: suggestion: avoid using internal styling

File: src/components/sidebar/categoryFilter.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  
+      const isCurrentChecked = e.target.checked;
+      setCheckedList((item) => {
+        const newCheckedList = item.map((obj) => {
+          if (obj.name === itemName) {
+            if (isCurrentChecked) return { ...obj, isTrue: true };
+            else return { ...obj, isTrue: false };
+          } else return { ...obj };
+        });
+        console.log(newCheckedList);
```
Comment (kgajriaAgarwal) [inline src/components/sidebar/categoryFilter.js]: remove console

File: src/pages/productPage/productPage.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import "../home/home.css";
+import { Footer, Navbar } from '../../components/allComponents';
+import "./productPage.css";
+import { ProductListing } from '../../components/productListing/productListing';
+import { Sidebar } from '../../components/sidebar/sidbar';
```
Comment (yaswanthmyneni) [inline src/pages/productPage/productPage.js]: suggestion: create index file in components and export all components from it

File: src/components/sidebar/sidbar.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  const [toggleSideBar,setToggleSideBar]=useState(false);
+
+  const filterBar=()=>{
+    setToggleSideBar(prev=>!prev)
+  }
+
+  const btnClearClickHandler = () => {
+
+    //setting filter input checked false
+    const elements = document.getElementsByTagName("input");
```
Comment (kgajriaAgarwal) [inline src/components/sidebar/sidbar.js:21]: avoid using document.getElementsByTagName

File: src/components/sidebar/sidbar.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    for (var i = 0; i < elements.length; i++) {
+      elements[i].checked = false;
+    }
+    //setting slider initial value
+    const element = document.getElementsByClassName("slider");
```
Comment (kgajriaAgarwal) [inline src/components/sidebar/sidbar.js:26]: avoid using document.getElementsByClassName

File: src/context/filterContext/filterContext.js [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+
+                if (isDataSorted.IsSorted) {
+                    if (isDataSorted.Sort === "HIGH_TO_LOW") {
+                        newState2 = highToLow(newState2);
+                    }
+                    if (isDataSorted.Sort === "LOW_TO_HIGH") {
+                        newState2 = lowToHigh(newState2);
+                    }
+                }
+                newState2 = filterByRating(newState2, Number(sliderRatingValue));
```
Comment (kgajriaAgarwal) [inline src/context/filterContext/filterContext.js:236]: improve the naming for better readabilty

File: src/components/sidebar/sidbar.js [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+import { CategoryFilter } from './categoryFilter';
+import { PriceFilter } from './priceFilter';
+import { SortLhHl } from './sort-lh-hl';
+
+
+
+function Sidebar() {
+  const { dispatch } = useFilter();
+  const [rating, setRating] = useState(5)
+  const [toggleSideBar,setToggleSideBar]=useState(false);
```
Comment (yaswanthmyneni) [inline src/components/sidebar/sidbar.js:12]: suggestion: keep state variables in the filter context.

File: src/context/filterContext/filterContext.js [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+    createContext,
+    useContext,
+    useEffect,
+    useReducer,
+    useState
+} from 'react'
+import axios from 'axios'
+
+
+const categories = [
```
Comment (kgajriaAgarwal) [inline src/context/filterContext/filterContext.js:11]: These categories should be managed from backend

**Review by kgajriaAgarwal — CHANGES_REQUESTED** [review]

**Review by yaswanthmyneni — APPROVED** [review]
you can merge it

File: src/components/sidebar/sidbar.js (hunk 9 above)
Comment (JuvitaSaini09) [reply src/components/sidebar/sidbar.js:21]: ok thanks fro the review

File: src/components/sidebar/sidbar.js (hunk 12 above)
Comment (JuvitaSaini09) [reply src/components/sidebar/sidbar.js:12]: ok thnks for the suggestion

File: src/pages/productPage/productPage.js (hunk 8 above)
Comment (JuvitaSaini09) [reply src/pages/productPage/productPage.js]: i have updated it

File: src/components/sidebar/categoryFilter.js (hunk 7 above)
Comment (JuvitaSaini09) [reply src/components/sidebar/categoryFilter.js]: updated,thnks for the review

**Pull request merged**